src/assignments/tech-jobs-persistent.rst

```diff
@@ -0,0 +1,382 @@
+``AbstractEntity``. This class is holds the fields and methods that are common across the ``Job`` class
+and the classes it contains as fields.  
+
+#. We will be creating tables for the subclasses that inherit from
+   ``AbstractEntity`` but not a table for this parent class. Therefore, give ``AbstractEntity`` the 
+   ``@MappedSuperClass`` annotation.
+
+#. Since all of the subclasses of ``AbstractEntity`` will be entities themselves, add the ``@Id`` 
+   and ``@GeneratedValue`` annotations to the field ``id``.
+
+#. Each subclass will also inherit the ``name`` field from ``AbstractEntity``. Add appropriate 
+   validation annotations so that:
+   
+   a. a user cannot leave this field blank when creating an object. 
+
+   b. there are reasonable limitations on the size of the name string.
+
+
+Models
+^^^^^^
+
+In the last assignment, a ``Job`` object contained string fields for employer and core competency data. This employer 
+and skill (formerly core competency) information about a particular job will now be stored in classes themselves.
+These items themselves will hold their own supplementary information. 
+
+#. Open the ``Employer`` model class. In addition to the fields inherited from ``AbstractEntity``, ``Employer`` should have a 
+   string field for ``location``. Add this field with validation as you see fit, as well as getters and setters.
+
+   .. admonition:: Note
+
+      For the purposes of this application, an employer can only have one location.
+
+#. ``Employer`` is a class that will be mapped to one of our tables. Make sure the class has the 
+   ``@Entity`` annotation, as well as the no-arg constructor required for Hibernate to create an
+   object.
+
+#. In the model class ``Skill``, add a field for a longer description of the skill. Some hiring managers like to have
+   more information available about the nature of a given programming language or framework. 
+
+#. As with ``Employer``, give this class the ``@Entity`` annotation and be sure it contains a no-arg
+   constructor.
+
+
+Data Layer
+^^^^^^^^^^
+
+To map the ``Employer`` and ``Skill`` classes to your techjobs database, you'll add data access interfaces for these relational 
+objects, similiar to the existing ``JobRepository`` interface. Like ``JobRepository``, make use of the Spring Data 
+``CrudRepository`` class to map our objects.
+
+#. In ``models/data``, create a new interface ``EmployerRepository``.
+
+   a. ``EmployerRepository`` should extend ``CrudRepository``.
+   #. ``EmployerRepository`` should be annotated with ``@Repository``.
+
+#. Repeat the steps above for an interface, ``SkillRepository``.
+
+Controllers
+^^^^^^^^^^^
+
+With the employer repository in place, we will reference this to send object information through 
+the ``EmployerController`` handlers. ``EmployerController`` contains two handlers with missing 
+information.
+
+#. View an individual employer object.
```

Comment:
This is a little disjointed.

Reply:
yeah i dont really know why thats there! going to consider it a typo